client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -89,6 +91,21 @@ function PatternPreviewFragment( {
 		copyButtonText = isPreviewLarge ? 'Pattern copied!' : 'Copied';
 	}
 
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+	const patternType =
+		typeof patternTypeFilter !== 'undefined'
+			? getTracksPatternType( patternTypeFilter )
+			: patternTypeFilter;
+	const recordCopyEvent = ( tracksEventName: string ) => {
+		recordTracksEvent( tracksEventName, {
+			name: pattern?.name,
+			category,
+			type: patternType,
+			user_is_dev_account: isDevAccount ? '1' : '0',
+			view: isGridView ? 'grid' : 'list',
+		} );
+	};
```

Comment:
```suggestion
	const recordCopyEvent = ( tracksEventName: string ) => {
		recordTracksEvent( tracksEventName, {
			name: pattern?.name,
			category,
			type: getTracksPatternType( patternTypeFilter ),
			user_is_dev_account: isDevAccount ? '1' : '0',
			view: isGridView ? 'grid' : 'list',
		} );
	};
```
I would rather copy @wongasy's changes to `getTracksPatternType` from here and make that function accept `undefined` too:

https://github.com/Automattic/wp-calypso/pull/88976/files#diff-2e7995a91953c3fadfc4a6045b28f8d5a840a5771d2bcbadd1e283db192fba53R3-R11

Reply:
agreed, this is much cleaner now, thanks!